target/linux/ath79/generic/base-files/etc/board.d/01_leds

```diff
@@ -134,6 +134,15 @@ comfast,cf-e314n-v2)
 	ucidef_set_led_rssi "rssimediumhigh" "RSSIMEDIUMHIGH" "green:rssimediumhigh" "wlan0" "51" "100"
 	ucidef_set_led_rssi "rssihigh" "RSSIHIGH" "green:rssihigh" "wlan0" "76" "100"
 	;;
+comfast,cf-e320n-v2)
+	ucidef_set_led_switch "lan" "LAN" "green:lan" "switch0" "0x04"
+	ucidef_set_led_switch "wan" "WAN" "red:wan" "switch0" "0x02"
+	ucidef_set_rssimon "wlan0" "200000" "1"
+	ucidef_set_led_rssi "rssilow" "RSSILOW" "red:rssilow" "wlan0" "1" "100"
+	ucidef_set_led_rssi "rssimediumlow" "RSSIMEDIUMLOW" "red:rssimediumlow" "wlan0" "26" "100"
+	ucidef_set_led_rssi "rssimediumhigh" "RSSIMEDIUMHIGH" "green:rssimediumhigh" "wlan0" "51" "100"
+	ucidef_set_led_rssi "rssihigh" "RSSIHIGH" "green:rssihigh" "wlan0" "76" "100"
```

Comment:
This is wrong. Did you not test it?

Reply:
Yes, it is, it's a common template for all the comfast devices. 

And I'm not sure what are you asking me to change in the DTS file. If you can please expand your answer. 

Thanks

Reply:
The names you use here do not exist in DTS. So, it obvious you just copy/pasted it without knowing how it works and without having tested it.

Reply:
> MAC address info in the commit message is also still missing: https://openwrt.org/docs/guide-developer/device-support-policies#mac_addresses
> 
> And please squash your changes into a single commit and force-push them to your branch, so this PR is updated.

```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel master br-lan state DOWN mode DEFAULT group default qlen 1000
    link/ether 20:0d:b0:87:3f:71 brd ff:ff:ff:ff:ff:ff
3: eth1: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN mode DEFAULT group default qlen 1000
    link/ether 20:0d:b0:87:3f:6f brd ff:ff:ff:ff:ff:ff
6: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DORMANT group default qlen 1000
    link/ether 20:0d:b0:87:3f:70 brd ff:ff:ff:ff:ff:ff
```

Can I assume it is like this ?

- eth0 = xx:71
- eth1 = xx:6f
- wlan0 = xx:70

I tried to find the info from the [original firmware](http://dl.comfast.cn/firmware/CF-E320NV2-V2.6.0.3.zip), but I didn't know how to extract the information from it.

Reply:
Typically, one would get the OEM assignment from a device with OEM firmware installed. Then, you could install OpenWrt on the device and look for the addresses in the usual locations (art 0x0, 0x6, 0x1002):
https://openwrt.org/docs/guide-developer/mac.address

This way you can relate the assignment from OEM to where you find these addresses, and finally compile the MAC address table I'm looking for. This is not at all possible without any OEM information.

One might also try to extract the same information from the OEM firmware, but it's not that easy.

Looking at your reported addresses here, you would have :6f for LAN and :71 for WAN, which would be the opposite compared to ar71xx. However, that doesn't help much, as we have three open questions related to that:
1. Is the assignment on ar71xx correct?
2. Is your assignment of LAN/WAN correct?
3. Is your setup of the &eth* DT nodes correct?

We do not have sufficient information yet to answer all these questions.

Unfortunately, the firmware download doesn't work for me either.